pulser-core/pulser/sequence/sequence.py

```diff
@@ -437,8 +475,102 @@ def config_slm_mask(self, qubits: Iterable[QubitId]) -> None:
         if self._slm_mask_targets:
             raise ValueError("SLM mask can be configured only once.")
 
-        # If checks have passed, set the SLM mask targets
         self._slm_mask_targets = targets
+        self._slm_mask_dmm = dmm_id
+        if self._in_xy and dmm_id not in self._device.dmm_channels:
```

Comment:
Doesn't the `dmm_id` need to be available too?
```suggestion
        if self._in_xy and dmm_id not in self.available_channels:
```

Reply:
At the moment, this is already checked above as if we are in xy mode, either no slm have been defined yet and then available_channels = dmm_channels and xy channels, or a slm has been defined but then _slm_mask_targets is defined and an error is raised before.

Reply:
I would rather not use `self.available_channels` because it uses `self._slm_mask_dmm`.